zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/Paragraph.java

```diff
@@ -61,6 +61,7 @@
   Date dateUpdated;
   private Map<String, Object> config; // paragraph configs like isOpen, colWidth, etc
   public final GUI settings;          // form and parameter settings
+  private Map<String, ParagraphRuntimeInfo> runtimeInfos;
```

Comment:
I see, thanks for explain. Do you see any other possible approaches? 
Because NotebookRepo designed to allow 3rd party implementation, and i think it's not the best making every possible NotebookRepo implementation handles the `runtimeInfos` with exclusionStrategy.